Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/CameraReelGalleryController.cs

```diff
@@ -0,0 +1,467 @@
+                Datetime = datetime;
+            }
+        }
+
+        public event Action<CameraReelResponseCompact>? ThumbnailClicked;
+        public event Action<CameraReelStorageStatus>? StorageUpdated;
+
+        private const int THUMBNAIL_POOL_DEFAULT_CAPACITY = 100;
+        private const int THUMBNAIL_POOL_MAX_SIZE = 10000;
+        private const int GRID_POOL_DEFAULT_CAPACITY = 10;
+        private const int GRID_POOL_MAX_SIZE = 500;
+        private const int ANIMATION_DELAY = 300;
+        private const int MINIMUM_ELEMENTS_FOR_WAITING_LAYOUT = 30;
+
+        private readonly CameraReelGalleryView view;
+        private readonly ICameraReelStorageService cameraReelStorageService;
+        private readonly IExplorePanelEscapeAction explorePanelEscapeAction;
+        private readonly ReelGalleryPoolManager reelGalleryPoolManager;
+        private readonly Dictionary<DateTime, MonthGridController> monthViews = new ();
+        private readonly Dictionary<CameraReelResponseCompact, Sprite> reelThumbnailCache = new ();
+        private readonly OptionButtonController optionButtonController;
+        private readonly ContextMenuController contextMenuController;
+        private readonly Rect elementMaskRect;
+
+        private bool isLoading = false;
+        private bool isDragging = false;
+        private float previousY = 1f;
+        private CancellationTokenSource loadNextPageCts = new ();
+        private CancellationTokenSource showSuccessCts = new ();
+        private CancellationTokenSource showFailureCts = new ();
+        private ReelThumbnailController[] thumbnailImages;
+        private int beginVisible;
+        private int endVisible;
+        private int currentSize;
+        private CameraReelResponseCompact reelToDelete;
+        private PagedCameraReelManager pagedCameraReelManager;
+
+        public CameraReelGalleryController(CameraReelGalleryView view,
+            ICameraReelStorageService cameraReelStorageService,
+            ICameraReelScreenshotsStorage cameraReelScreenshotsStorage,
+            IWebBrowser webBrowser,
+            IDecentralandUrlsSource decentralandUrlsSource,
+            IExplorePanelEscapeAction explorePanelEscapeAction,
+            OptionButtonView optionButtonView = null,
+            ContextMenuView contextMenuView = null)
+        {
+            this.view = view;
+            this.cameraReelStorageService = cameraReelStorageService;
+            this.explorePanelEscapeAction = explorePanelEscapeAction;
+            this.view.Disable += OnDisable;
+            this.view.scrollRectDragHandler.BeginDrag += ScrollBeginDrag;
+            this.view.scrollRectDragHandler.EndDrag += ScrollEndDrag;
+            this.view.scrollBarDragHandler.BeginDrag += ScrollBeginDrag;
+            this.view.scrollBarDragHandler.EndDrag += ScrollEndDrag;
+            this.elementMaskRect = this.view.elementMask.GetWorldRect();
+
+            if (optionButtonView is not null && contextMenuView is not null)
```

Comment:
there is no use case for that in this PR but this check exist since this component will be reused in the Passport and in the navmap where the context menu and options button don't exist.
Those views are also under a "Nullable References" header (for the inspector) to indicate that the prefab can be used without those components